include/PathOrderOptimizer.h

```diff
@@ -421,53 +430,65 @@ class PathOrderOptimizer
                 return nullptr;
             };
 
-
-        // We want all outer walls to be printed first (For outside in ordering) or last (for inside out ordering)
-        std::unordered_set<Path> outer_walls = reverse_direction ? leaves : roots;
-
-        if (reverse_direction)
+        if (group_outer_walls)
         {
-            visited.insert(outer_walls.begin(), outer_walls.end());
+            // We want all outer walls to be printed first (For outside in ordering) or last (for inside out ordering)
+            std::unordered_set<Path> outer_walls = reverse_direction ? leaves : roots;
```

Comment:
Minor remark but I would place this in the individual branches of the next if-statement. I think it would improve readability of the code as you can write
```cpp
if (reverse_direction)
{
   // as the order is reverse the leaves of the order requirement are the outer walls
   std::unordered_set<Path> outer_walls = leaves;
   .....
}
else
{
   // as the order is not reverse the roots of the order requirement are the outer walls
   std::unordered_set<Path> outer_walls = roots;
   .....
}
```